strawberryfields/apps/graph/resize.py

```diff
@@ -363,11 +363,13 @@ def clique_shrink(subgraph: list, graph: nx.Graph) -> list:
     if not utils.is_subgraph(subgraph, graph):
         raise ValueError("Input is not a valid subgraph")
 
-    subgraph = graph.subgraph(subgraph).copy()  # Create NetworkX Graph version of subgraph
+    subgraph = graph.subgraph(subgraph).copy()  # A copy is required to be able to modify structure
 
```

Comment:
Done!

Reply:
Sorry for the last minute objection, but can we drop this? We shouldn't need this level of documentation for a common python pattern.

Reply:
I would be happy to drop it completely, but happy to go with the decision of the reviewers @co9olguy @nquesada who commented on it earlier?

Reply:
I know it seems trivial but this is a bit of a principled objection for me--not just Python but all languages implement some sort of pattern to deal with mutable data structures. It's an everyday kind of problem. If we commented everything this way the code would quickly become unreadable.

Reply:
Is it a common pattern? When reviewing code, I wondered why a `copy` was necessary on an otherwise newly created object. It was certainly not clear to me why it was necessary. I understand not commenting common patterns, but sometimes comments can remove friction for future reviewers of the code

Reply:
Nobody is asking to comment everything like this. Just the cases where there is genuine confusion from actual developers

Reply:
Fair points but I still strongly disagree with this. Would prefer to discuss in person if it's a sticking point.

Reply:
Fair point but I still disagree with this. If it's confusing, better to rewrite the code in a way that makes things understandable without the use of a comment.

From Robert Martin's Clean Code:

> The proper use of comments is to compensate for our failure to express ourself in code. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration. So when you find yourself in a position where you need to write a comment, think it through and see whether there isn’t some way to turn the tables and express yourself in code.

This could solve the issue if it's confusing:

```python
nx_subgraph_view = graph.subgraph(subgraph)
nx_subgraph = nx_subgraph_view.copy()
```